activerecord/test/cases/relation/where_clause_test.rb

```diff
@@ -74,6 +74,33 @@ class WhereClauseTest < ActiveRecord::TestCase
       assert_equal expected, a.merge(b)
     end
 
+    test "merge with or/and correctly handles SQL literals and attributes" do
+      a = WhereClause.new([
+        table["id"].eq(1),
+        Arel::Nodes::Grouping.new(Arel.sql("foo = bar"))
+      ])
+      b = (a + WhereClause.new([table["attr_1"].eq("val_1")])).or(
+        WhereClause.new([table["attr_2"].eq("val_2")])
+      )
+
+      expected = WhereClause.new([
+        table["id"].eq(1),
+        Arel::Nodes::Grouping.new(Arel.sql("foo = bar")),
+        Arel::Nodes::Grouping.new(
+          Arel::Nodes::Or.new([
+            Arel::Nodes::And.new([
+              table["id"].eq(1),
+              Arel::Nodes::Grouping.new(Arel.sql("foo = bar")),
+              table["attr_1"].eq("val_1")
+            ]),
+            table["attr_2"].eq("val_2")
+          ])
+        )
+      ])
+
+      assert_equal expected, a.merge(b)
```

Comment:
Without patch:
![Screenshot 2025-01-09 at 7 38 56 pm](https://github.com/user-attachments/assets/1ae7be48-f8d0-4cb2-a13a-f5cb124cfad2)

The matching expectation for the failure would be:

```ruby
      expected = WhereClause.new([
        # table["id"].eq(1),
        Arel::Nodes::Grouping.new(Arel.sql("foo = bar")),
        Arel::Nodes::Grouping.new(
          Arel::Nodes::Or.new([
            Arel::Nodes::And.new([
              table["id"].eq(1),
              Arel::Nodes::Grouping.new(Arel.sql("foo = bar")),
              table["attr_1"].eq("val_1")
            ]),
            table["attr_2"].eq("val_2")
          ])
        )
      ])
```